### PR TITLE
[Flask] remove snap data from file system after install

### PR DESCRIPTION
**Pull request by owencraston**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
- after a snap was installed, it would leave the archived and unarchives source code on the users file system. 
- This is not good for many reasons, it takes up a lot of the users storage
- it also indicates to the app store reviewers tat we write data to the file system and do not clean it up
_2. What is the improvement/solution?_
- after we are done with the request, we flush the response, this will cleanup the archived source code (.tgz file)
- after a successful installation we pass the data path to a cleanup function and we delete that folder too. This folder holes the unarchived source code. 

## Testing
iOS
- download build #1110 v6.6.1 from testflight
- install a snap from either the snaps dev menu or the browser via tss.ac
- go to the file system and verify that no new file files were added. Look out for the files/folders with the prefix RNFetchBlob...

Android
- download and install the apk from [here](https://app.bitrise.io/build/85056d57-c06a-45a0-9bee-ba51872bcfbc?tab=artifacts)
- download and install a snap
- verify that no new files were added to the file system after a successful install. 
 
**Screenshots/Recordings**

before:
Screenshot_1684347483

After:
Screenshot_20230517-110141

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/776

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/core/Snaps/location/npm.ts [hunk 1]
```diff
@@ -90,10 +90,7 @@ const fetchAndStoreNPMPackage = async (
   inputRequest: RequestInfo,
 ): Promise<string> => {
   const { config } = RNFetchBlob;
-  const targetDir =
-    Platform.OS === 'ios'
-      ? RNFetchBlob.fs.dirs.DocumentDir
-      : `${RNFetchBlob.fs.dirs.DownloadDir}`;
+  const targetDir = RNFetchBlob.fs.dirs.DocumentDir;
```
Comment (owencraston) [inline app/core/Snaps/location/npm.ts:93]: I determined that this using the download directory was not needed on Android and actually blocked me from deleting the files after.

Comment (owencraston) [reply app/core/Snaps/location/npm.ts:93]: Most likely a permissions related issue. Its much easier to just use documents anyway.

File: android/app/build.gradle [hunk 2]
```diff
@@ -159,8 +159,8 @@ android {
 		applicationId "io.metamask"
 		minSdkVersion rootProject.ext.minSdkVersion
 		targetSdkVersion rootProject.ext.targetSdkVersion
-		versionCode 1104
-		versionName "6.5.0"
+		versionCode 1110
```
Comment (owencraston) [inline android/app/build.gradle:162]: will be removed before merging.

File: bitrise.yml [hunk 3]
```diff
@@ -620,10 +620,10 @@ app:
       PROJECT_LOCATION_IOS: ios
     - opts:
         is_expand: false
-      VERSION_NAME: 6.5.0
+      VERSION_NAME: 6.6.1
```
Comment (owencraston) [inline bitrise.yml:623]: will be removed before merging.

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1154,7 +1154,7 @@
 				CODE_SIGN_ENTITLEMENTS = MetaMask/MetaMaskDebug.entitlements;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Manual;
-				CURRENT_PROJECT_VERSION = 1104;
+				CURRENT_PROJECT_VERSION = 1110;
```
Comment (owencraston) [inline ios/MetaMask.xcodeproj/project.pbxproj:1157]: will be removed before merging.

**Review by gantunesr — APPROVED** [review]
LGTM

**Pull request merged**